### ISSUE TITLE
Bug: Node Transforms of meshes loaded from collada file doesn't match with file

### ISSUE THREAD

**Issue by barisdoga2** [assimp/assimp#4543]
**Quick Description**
I am trying to reach the location data of meshes but they come wrong. The transformation matrix doesn't deformed like it is prepared. If no rotation applied on the meshes the location data is correct.

**To Reproduce**
1.Load a collada file.
2.Call aiMatrix4x4::Decompose for every child of root node on the scene.
3.Check the location buffer, If object has any rotation, the resulted value will not match the one on the file.

**Expected behavior**
Location vectors at collada files, should be able to be decomposable exactly like on the life.

**Explanation**
Collada File look like this, 45 degrees rotation on x axix
```
  <library_visual_scenes>
    <visual_scene id="Scene" name="Scene">
      <node id="Cube_002" name="Cube.002" type="NODE">
        <scale sid="scale">1 1 1</scale>
        <rotate sid="rotationZ">0 0 1 0</rotate>
        <rotate sid="rotationY">0 1 0 0</rotate>
        <rotate sid="rotationX">1 0 0 45</rotate>
        <translate sid="location">50.28152 254.1107 -398.2061</translate>
```

- While assimp is loading the mesh, it hits to `aiMatrix4x4 ColladaParser::CalculateResultTransform(const std::vector<Transform>& pTransforms) const` function. Inside of this function 5 different transforms are applied to the node, exactly in the following order scale, rotz, roty, rotx, location.  Lets consider scale as 1 and other rotations as 0 except rotx. The function body looks like;

```
        case TF_ROTATE:
        {
            aiMatrix4x4 rot;
            ai_real angle = tf.f[3] * ai_real(AI_MATH_PI) / ai_real(180.0);
            aiVector3D axis(tf.f[0], tf.f[1], tf.f[2]);
            aiMatrix4x4::Rotation(angle, axis, rot);
            res *= rot;
            break;
        }
        case TF_TRANSLATE:
        {
            aiMatrix4x4 trans;
            aiMatrix4x4::Translation(aiVector3D(tf.f[0], tf.f[1], tf.f[2]), trans);
            res *= trans;
            break;
        }
```

- So at the end of the function location is multiplied with result matrix and the resulted value for position will be: vec3(50,-461,101)

- Also want to mention about Decompose method implementation, it looks like it is directly geting the location from matrix indices, doesn't do any arithmetics.

```
#define ASSIMP_MATRIX4_4_DECOMPOSE_PART		\
	const aiMatrix4x4t<TReal>& _this = *this;/* Create alias for conveniance. */ \
	\
	/* extract translation */ \
	pPosition.x = _this[0][3]; \
	pPosition.y = _this[1][3]; \
	pPosition.z = _this[2][3]; \
	\
```

**Platform**
 - Blender Export: 
 - Global Orientation,  Forward axis: -Z, Up axis: Y, Geometry->transform->decompose
 - OS: Windows 32bit
 - Assimp latest, vcpkg

Thanks!

**Comment by RichardTea** [comment]
Here be dragons!

It is mathematically impossible to round-trip TRS -> Matrix4x4 -> TRS (except for identity)
This is because there are an infinite number of TRS solutions for a given Matrix4x4.
Every decomposition algorithm has a slightly different strategy, so will prefer a different decomposition that may be very different to the TRS that originally created it.

Euler angles make this even worse, as there are an infinite number of equivalent Euler orientations.

TL;DR: TRS numbers **will** change. Do not compare them!

That said, it is possible that the Collada importer is applying the transform steps in an incorrect order.

To figure that out, you'll need to export the same scene from 3D authoring tool using in both this Euler TRS and Matrix4x4 formats, and see if assimp imports them the same.

PS: That is genuinely how you extract position from a Matrix4x4
